### PR TITLE
fix: register the `ConstraintWeightOverrides` access member for native compilation

### PR DESCRIPTION
**Pull request by zepfred**
This pull request adds the `ConstraintWeightOverrides` access member for native compilation.

### CODE REVIEW

**Comment by zepfred** [comment]
@triceo, I've confirmed the issue does not affect the Spring boot native build.

File: quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java [hunk 1]
```diff
@@ -920,6 +907,32 @@ private GeneratedGizmoClasses generateDomainAccessors(Map<String, SolverConfig>
...
+            // The ConstraintWeightOverrides field is not annotated, but it needs a member accessor
+            AnnotationInstance solutionClassInstance = planningSolutionAnnotationInstanceCollection.iterator().next();
+            ClassInfo solutionClassInfo = solutionClassInstance.target().asClass();
+            FieldInfo constraintFieldInfo = solutionClassInfo.fields().stream()
+                    .filter(f -> f.type().name().equals(DotNames.CONSTRAINT_WEIGHT_OVERRIDES))
+                    .findFirst()
+                    .orElse(null);
+            if (constraintFieldInfo != null) {
+                // Prefer method to field
+                // (In Python, the field doesn't implement the interface).
```
Comment (Christopher-Chianelli) [inline quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java:919]: Python cannot use Quarkus, so this comment is worthless here
```suggestion
```

**Review by Christopher-Chianelli — APPROVED** [review]
The Python comment can be removed from the Quarkus processor, since you cannot use Quarkus with Python. LGTM.

**Comment by triceo** [comment]
Also, the "new issues" identified by Sonar are easily fixable.

**Pull request merged**